src/utils/url.ts

```diff
@@ -0,0 +1,13 @@
+export function normalizeProtocolUrl(url: string): string {
+  const fullUrl = url.includes('://') ? url : `http://${url}`;
```

Comment:
what if the URL contains `://` in the path/query/etc.?  perhaps we should `/^[^:]+:\/\/./.test(url)` instead?